internal/gps/pkgtree/pkgtree.go

```diff
@@ -445,6 +450,20 @@ func (t PackageTree) ToReachMap(main, tests, backprop bool, ignore map[string]bo
 		ignore = make(map[string]bool)
 	}
 
+	// Declare a radix tree for ignore prefixes
+	var xt *radix.Tree
+
+	for i := range ignore {
+		if strings.HasSuffix(i, wcIgnoreSuffix) {
+			// Create trie if it doesn't exists
+			if xt == nil {
+				xt = radix.New()
+			}
+			i = strings.TrimSuffix(i, wcIgnoreSuffix)
```

Comment:
This ends up on the `strings.HasSuffix()` call. better to just do the trim operation ourselves.

Reply:
You mean for cases like `"github.com/x/someprefix-*-somesuffix"`?
Like using regex match?

Reply:
nope, i meant that this is the implementation of `string.TrimSuffix()`:

```go
func TrimSuffix(s, suffix string) string {
	if HasSuffix(s, suffix) {
		return s[:len(s)-len(suffix)]
	}
	return s
}
```

and the conditional has already been checked, so it's better to just do the `s[:len(s)-len(suffix)]` operation ourselves rather than calling `strings.TrimSuffix()`, as this method is processing-bound (possibly memory, possibly CPU), and this is within a loop, so easy microoptimizations like this are worth it.